assembly/src/assembler/instruction/procedures.rs

```diff
@@ -147,11 +58,15 @@ impl Assembler {
         callee: &InvocationTarget,
         proc_ctx: &mut ProcedureContext,
         basic_block_builder: &mut BasicBlockBuilder,
-        mast_forest_builder: &MastForestBuilder,
+        mast_forest_builder: &mut MastForestBuilder,
     ) -> Result<(), AssemblyError> {
-        let digest =
-            self.resolve_target(InvokeKind::ProcRef, callee, proc_ctx, mast_forest_builder)?;
-        self.procref_mast_root(digest, basic_block_builder)
+        let mast_root = {
+            let proc_body_id =
+                self.resolve_target(InvokeKind::ProcRef, callee, proc_ctx, mast_forest_builder)?;
+            mast_forest_builder.get_mast_node(proc_body_id).unwrap().digest()
```

Comment:
nit: could we add a comment here explaining why `unwrap()` is OK?

Reply:
I added a comment but honestly the `unwrap()` is true by construction: `MastNodeId` is only created by a `MastForest`, so all `MastNodeId`s are associated with a `MastNode` in a `MastForest` (unless you use it with the wrong mast forest or whatnot, but that's an internal error where I believe a `panic!` is appropriate).

We used to implement `Index` on `MastForestBuilder` for this purpose, and I'm not sure why we removed it

Reply:
Yeah I was surprised we removed `Index` as well, since I thought it was more-common-than-not that we were accessing node data from the forest, by id, with an id that we know a priori is in the forest, such as this case. Might be worth adding it back if the thinking was that it wasn't being used.